### ISSUE TITLE
Python 3.7 is not supported

### ISSUE THREAD

**Issue by ahrib** [nabla-c0d3/sslyze#325]
Microsoft Windows [Version 10.0.17134.137]
Python 3.7.0

pip install --upgrade sslyze

*result*
Collecting sslyze
  Using cached https://files.pythonhosted.org/packages/53/67/c8dc163326a4cafc31e2b37a11090d6d98e4811b8820aac87452d82dd205/SSLyze-1.4.3.tar.gz
Collecting nassl<1.2.0,>=1.1.0 (from sslyze)
  Could not find a version that satisfies the requirement nassl<1.2.0,>=1.1.0 (from sslyze) (from versions: 0.13.4.win32, 0.13.1, 0.13.2, 0.13.4, 0.13.5, 0.13.6, 0.13.7, 0.14.0, 0.14.1, 0.14.2, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.16.2, 0.16.3, 0.17.0, 1.0.1, 1.0.2, 1.0.3)
No matching distribution found for nassl<1.2.0,>=1.1.0 (from sslyze)

*expected*
normal installation

*workarounds attempted*

1. pip install --upgrade sslyze==1.4.3 
result: fail, same error

2. pip install --upgrade sslyze==1.4.2 
result: fail, same error

3. pip install --upgrade sslyze==1.4.1 
result: fail, same error

4. pip install --upgrade sslyze==1.4.0 
result: fail, same error

5. pip install --upgrade sslyze==1.3.4 
result: fail, Command "c:\users\a\appdata\local\programs\python\python37-32\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\A\\AppData\\Local\\Temp\\pip-install-gzdo7wfs\\nassl\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\A\AppData\Local\Temp\pip-record-3cjcgi65\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\A\AppData\Local\Temp\pip-install-gzdo7wfs\nassl\

**Comment by kjbeeman** [comment]
Just another user here who has been fighting this for the past hour or so. 

there isn't a version of Nassl available for Python3.7.  1.0.3 is the latest version that Python3.7 will install. 

Work around for now: use python3.6

I started looking into building Nassl for Python3.7, but my docker image was based on alpine and well, ran into other issues before I pulled ubuntu and python3.6 

Good luck.

**Comment by nabla-c0d3** [comment]
Python 3.7 is not supported; it will be in the next release.

**Comment by lazorgator** [comment]
I am also experiencing this with Python 3.5.3 under PyPy. The version pip shows available that would fit this requirement is unavailable.

```
[latrans@canis ~]$ python3 --version
Python 3.7.0
[latrans@canis ~]$ pypy3 --version
Python 3.5.3 (fdd60ed87e941677e8ea11acf9f1819466521bf2, Jun 22 2018, 02:39:05)
[PyPy 6.0.0 with GCC 8.1.1 20180531]
[latrans@canis ~]$ pypy3 -m venv .venv
[latrans@canis ~]$ source .venv/bin/activate
(.venv) [latrans@canis ~]$ python --version
Python 3.5.3 (fdd60ed87e941677e8ea11acf9f1819466521bf2, Jun 22 2018, 02:39:05)
[PyPy 6.0.0 with GCC 8.1.1 20180531]
(.venv) [latrans@canis ~]$ pypy3 --version
Python 3.5.3 (fdd60ed87e941677e8ea11acf9f1819466521bf2, Jun 22 2018, 02:39:05)
[PyPy 6.0.0 with GCC 8.1.1 20180531]
(.venv) [latrans@canis ~]$ pip install sslyze
Collecting sslyze
  Cache entry deserialization failed, entry ignored
  Using cached https://files.pythonhosted.org/packages/53/67/c8dc163326a4cafc31e2b37a11090d6d98e4811b8820aac87452d82dd205/SSLyze-1.4.3.tar.gz
Collecting nassl<1.2.0,>=1.1.0 (from sslyze)
  Cache entry deserialization failed, entry ignored
  Could not find a version that satisfies the requirement nassl<1.2.0,>=1.1.0 (from sslyze) (from versions: 0.13.4.win32, 0.13.1, 0.13.2, 0.13.4, 0.13.5, 0.13.6, 0.13.7, 0.14.0, 0.14.1, 0.14.2, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.16.2, 0.16.3, 0.17.0, 1.0.1, 1.0.2, 1.0.3)
No matching distribution found for nassl<1.2.0,>=1.1.0 (from sslyze)
You are using pip version 9.0.1, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(.venv) [latrans@canis ~]$ pip install --upgrade pip
Collecting pip
  Using cached https://files.pythonhosted.org/packages/5f/25/e52d3f31441505a5f3af41213346e5b6c221c9e086a166f3703d2ddaf940/pip-18.0-py2.py3-none-any.whl
Installing collected packages: pip
  Found existing installation: pip 9.0.1
    Uninstalling pip-9.0.1:
      Successfully uninstalled pip-9.0.1
Successfully installed pip-18.0
(.venv) [latrans@canis ~]$ pip install sslyze
Collecting sslyze
  Using cached https://files.pythonhosted.org/packages/53/67/c8dc163326a4cafc31e2b37a11090d6d98e4811b8820aac87452d82dd205/SSLyze-1.4.3.tar.gz
Collecting nassl<1.2.0,>=1.1.0 (from sslyze)
  Could not find a version that satisfies the requirement nassl<1.2.0,>=1.1.0 (from sslyze) (from versions: 0.13.4.win32, 0.13.1, 0.13.2, 0.13.4, 0.13.5, 0.13.6, 0.13.7, 0.14.0, 0.14.1, 0.14.2, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.16.2, 0.16.3, 0.17.0, 1.0.1, 1.0.2, 1.0.3)
No matching distribution found for nassl<1.2.0,>=1.1.0 (from sslyze)
(.venv) [latrans@canis ~]$ 
```

**Comment by nabla-c0d3** [comment]
Pypy is not supported as sslyze has a native/C module (https://github.com/nabla-c0d3/nassl).

**Comment by nabla-c0d3** [comment]
Python 3.7 is now available as part of [2.0.0](https://github.com/nabla-c0d3/sslyze/releases/tag/2.0.0).
